server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -502,21 +502,23 @@ public void beforeIndexRemoved(
                                     new IndicesModuleRequest(indexModule),
                                     acknowledgedResponseHandler
                                 );
-                                inProgressIndexNameFuture.whenComplete((r, e) -> {
-                                    if (e != null) {
-                                        inProgressFuture.complete(response);
-                                    } else if (e == null) {
-                                        inProgressFuture.completeExceptionally(e);
-                                    }
-                                });
-                            } catch (Exception ex) {
-                                inProgressFuture.completeExceptionally(ex);
+                                inProgressIndexNameFuture.orTimeout(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS).join();
+                            } catch (CompletionException e) {
+                                if (e.getCause() instanceof TimeoutException) {
+                                    logger.info("No response from extension to request.");
+                                }
+                                if (e.getCause() instanceof RuntimeException) {
```

Comment:
curious why this isn't an else if like line 512.

Reply:
Not sure - I should fix that

Reply:
Actually I think it's intentional, because no exception handling happens in the first if statement: there's only an additional logging statement.  The other if/else ifs actually handle the exception.

Reply:
Ah, I see.  A timeout _is_ a runtime exception so we log _and_ throw.